keymaster/ta/mbedtls_proxy.c

```diff
@@ -225,9 +223,9 @@ static TEE_Result mbedTLS_import_ecc_pk(mbedtls_pk_context *pk,
 		goto out;
 	}
 
-	if ((mbedtls_ret = mbedtls_mpi_copy(&ecc->Q.X, &attrs[0]) != 0) ||
-	    (mbedtls_ret = mbedtls_mpi_copy(&ecc->Q.Y, &attrs[1]) != 0) ||
-	    (mbedtls_ret = mbedtls_mpi_copy(&ecc->d, &attrs[2]) != 0) ||
+	if ((mbedtls_ret = mbedtls_mpi_copy(&ecc->Q.X, &attrs[1]) != 0) ||
+	    (mbedtls_ret = mbedtls_mpi_copy(&ecc->Q.Y, &attrs[2]) != 0) ||
+	    (mbedtls_ret = mbedtls_mpi_copy(&ecc->d, &attrs[0]) != 0) ||
```

Comment:
@omasse06 If you don't change the order of the EC key attributes as mentioned in https://github.com/linaro-swg/kmgk/pull/9/files/f9d6cdf35d1c2dec5d722310b6c2929cd1478093..a82eb8df6079cd523cd811fd6f23d07b40486935#r284733434, then perhaps you don't have to make this change as well?

Reply:
@vchong We used PKCS8 to decode private RSA and ECDSA provisioned keys.
As described in https://tools.ietf.org/html/rfc5915 the PKCS8 private key format is as follow:
PrivateKeyInfo ::= SEQUENCE {
        version                   Version,
        privateKeyAlgorithm       PrivateKeyAlgorithmIdentifier,
        privateKey                PrivateKey,
        attributes           [0]  IMPLICIT Attributes OPTIONAL }

where PrivateKey is the TEE_ATTR_ECC_PRIVATE_VALUE
and attributes are the public key value : (TEE_ATTR_ECC_PUBLIC_VALUE_X, TEE_ATTR_ECC_PUBLIC_VALUE_Y)

Reply:
@omasse06 Thanks! I guess maybe the order doesn't matter as much since it seems like each member array is always handled separated rather than together as a consecutive series or group of elements.